go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -546,13 +571,13 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 			pod.Namespace, pod.Name, pod.Spec.NodeName, podState)
 	}
 
-	ls, err = bnc.waitForNodeLogicalSwitch(switchName)
+	ls, err = bnc.waitForNodeLogicalSwitch(switchNames.Current)
```

Comment:
> I wondered if we should also check for the original switch.

This is only possible if this is non interconected or the zone has multiple nodes, for zones with just one node the switch manager only knows about the current switch.

> 
> What happens if the original switch/node is removed from the cluster?

If this is happening after live migration the VM will have an "orphan" IP that's not an issue since IP get copied during live migrations, if later on a new node/switch pop up and the VMs subnet is assigned the node created event is received at remote zone and  VM's ip is reserved again at the switch manager there so no new pods are going to have it.

> 
> Are we just going to have errors when we process migrated pod events? Will the pod not be able to migrate again?
> 

It will be.

> Will migrated pods only work while their original node is in the cluster?

Should not.